portfolyo/core/pfstate/pfstate_helper.py

```diff
@@ -67,7 +68,15 @@ def prepare_unsourcedprice(unsourcedprice: Any, ref_idx: pd.DatetimeIndex) -> Pf
 
 def prepare_sourced(sourced: Any, ref_idx: pd.DatetimeIndex) -> PfLine:
     if sourced is None:
-        return create.flatpfline(pd.DataFrame({"q": 0, "r": 0}, ref_idx))
+        return create.flatpfline(
+            pd.DataFrame(
+                {
+                    "q": [Q_(0, "MWh") for _ in range(len(ref_idx))],
```

Comment:
Here it's probably faster to do `pd.DataFrame({'q': pd.Series(0, ref_idx, dtype='pint[MWh]', 'r': pd.Series(0, ref_idx, dtype='pint[Eur]'})`

Reply:
not so easy to adapt tests to this change, didn't have time to change